bin/runtime-common/src/messages.rs

```diff
@@ -385,12 +387,18 @@ pub mod target {
 		}
 
 		fn dispatch(message: DispatchMessage<Self::DispatchPayload, BalanceOf<BridgedChain<B>>>) {
+			let message_id = (message.key.lane_id, message.key.nonce);
 			if let Ok(payload) = message.data.payload {
 				pallet_bridge_call_dispatch::Module::<ThisRuntime, ThisCallDispatchInstance>::dispatch(
```

Comment:
I know understand why we need to decode the message before this point (getting weight info), but I still don't like the explicit `note_rejected_message` call, could we just pass the entire `message.data.payload` (instead of `payload.0`) to the `dispatch` function?

I.e. I'm proposing to change the `fn dispatch` function from the trait to take `Result<...>` instead of just the payload - this emitting the event can stay internal to `call-dispatch`.